src/ReactPlaceholder.tsx

```diff
@@ -1,128 +1,142 @@
 import * as React from 'react';
-import * as PropTypes from 'prop-types';
 import * as placeholders from './placeholders';
+import { joinClassNames } from './utils';
 
 export type CommonProps = {
-  children: React.ReactNode,
+  children: React.ReactElement | null;
   /** pass `true` when the content is ready and `false` when it's loading */
-  ready: boolean,
+  ready: boolean;
   /** delay in millis to wait when passing from ready to NOT ready */
-  delay?: number,
+  delay?: number;
   /** if true, the placeholder will never be rendered again once ready becomes true, even if it becomes false again */
-  firstLaunchOnly?: boolean,
-  className?: string,
-  style?: React.CSSProperties
-}
-
-export type Props = (CommonProps & {
-  /** type of placeholder to use */
-  type: 'text' | 'media' | 'textRow' | 'rect' | 'round',
-  /** number of rows displayed in 'media' and 'text' placeholders */
-  rows?: number,
-  /** color of the placeholder */
-  color?: string,
-  /** pass true to show a nice loading animation on the placeholder */
-  showLoadingAnimation?: boolean,
-  customPlaceholder?: undefined
-}) | (CommonProps & {
+  firstLaunchOnly?: boolean;
+  className?: string;
+  style?: React.CSSProperties;
+};
+
+export type PlaceholderProps = CommonProps & {
+  // we have a default color, so we can set this as optional
+  color?: string;
+  // we have a default color, so we can set this as optional
+  rows?: number;
+  showLoadingAnimation?: boolean;
+  customPlaceholder?: undefined;
+};
+
+export type CustomPlaceholderProps = CommonProps & {
```

Comment:
should this be exported? What's the need?

Reply:
Good call, I removed exporting all of the prop interfaces except for the top-level one.